src/xml/XMLAttribute.hpp

```diff
@@ -155,13 +155,13 @@ void XMLAttribute<ATTRIBUTE_T>::readValue(const std::map<std::string, std::strin
 {
   PRECICE_TRACE(_name);
   if (_read) {
-    PRECICE_ERROR("Attribute \"" + _name + "\" is defined multiple times");
+    PRECICE_ERROR("Attribute \"" + _name + "\" has already been defined. Duplications are not permitted.");
   }
 
   const auto position = aAttributes.find(getName());
   if (position == aAttributes.end()) {
     if (not _hasDefaultValue) {
-      PRECICE_ERROR("Attribute \"" + _name + "\" missing");
+      PRECICE_ERROR("Attribute \"" + _name + "\" is required, but was not defined.");
```

Comment:
It would be good if here we also had some context: The user only gets a message that "you also need X", without defining where. As often this would not happen because of a simple typo (but because something is really missing), this would be tricky to debug.

Reply:
I was thinking about this as well. The problem is that there is no access to the tag from the attribute itself.
So this needs additional work.

Reply:
I opened a specific issue (https://github.com/precice/precice/issues/827), let's skip this for now.